### PR TITLE
Problem 5 Submission - ADNI Classification using GFNet 

### PR DESCRIPTION
**Pull request by Ens-on-matt**
This is my pull request for my implementation of the vision transformer GFNet to classify ADNI brain scans for Alzheimer's (AD) or Normal Cognition (NC). 
The files train.py, test.py, dataset.py and modules.py are contained alongside a README documenting the model, process and how to use train.py to train the model and predict.py to test its performance.
Unfortunately the model does not reach the goal accuracy (0.8) but I have documented the steps I have taken to attempt to improve the model and their effectiveness.

### CODE REVIEW

**Comment by wangzhaomxy** [comment]
**<This is an initial inspection, no action is required at this point.>**

**File Organizing:** Well-organized files.

**Problem Solving:**

- The algorithm solves the problem appropriately. 
- Accuracy in testing dataset: 0.629. 

**Model and functions:**

- It correctly uses PyTorch to construct the GFNet models and functions.
- With data augmentation.

**Code design:** Good.

**Code comment and docstring:**

- Good code comments 
- Good function docstrings, but it would be better if using the classical Python style.

**Difficulty:** Hard.

**Additional Comments:**

- Good exploration of the experiment.
- Versions information should be involved in the dependencies.
- Reasonable commits
- Good ReadMe design, but the GFNet paper should be involved in the Reference instead of Acknowledgments. In addition, it would be better if there is an independ conclusion part to summarize your report.

**Comment by aniketgupta17** [comment]
Observational Feedback

Pull Request:
Correctly created the Pull request from Topic Recognition Branch .
The pull request shows no incorporation of previous feedback .
The pull request should have a description of the file structure . 

File Organizing: Well-organized files.

Commit Log:
Commit messages are meaningful .
Commits demonstrate a logical development flow.
Commits are regularly made.

Documentation:
The README file provides a clear explanation of the algorithm and the problem it solves.
Code comments and docstrings are included .
Proper GitHub markdown formatting is used, with organized headings, lists, and code blocks.
Can add about Conclusion and Future works for  the report

**Comment by hanemma7moud** [comment]
Matthew Haley

**Comment by gayanku** [comment]
## Marking
<table><tbody>
  <tr><td colspan='3'><b>Good/OK/Fair Practice (Design/Commenting, TF/Torch Usage)</b></td> </tr>
	<tr><td></td><td>Good design and implementation. </td><td></td></tr>
	<tr><td></td><td>Spacing and comments. </td><td></td></tr>
	<tr><td></td><td>No Header blocks. </td><td>-1</td></tr>
	
  <tr><td colspan='3'><b>Recognition Problem</b></td></tr>
	<tr><td></td><td>OK solution to problem. Accsomewhat low.</td><td>-1</td></tr>
	<tr><td></td><td>Driver Script present. </td><td></td></tr>
	<tr><td></td><td>File structure present. </td><td></td></tr>
	<tr><td></td><td>Good Usage & Demo & Visualisation & Data usage. </td><td></td></tr>
	<tr><td></td><td>Module present. </td><td></td></tr>
	<tr><td></td><td>Commenting present. </td><td></td></tr>
	<tr><td></td><td>No Data leakage found. Uses Data Aug.</td><td></td></tr>
	<tr><td></td><td>Difficulty : Hard.  GFNet (Hard Difficulty)</td><td></td></tr>
	
  <tr><td colspan='3'><b>Commit Log</b></td></tr>
	<tr><td></td><td>Some/Adequate Meaningful commit messages. </td><td>-1</td></tr>
	<tr><td></td><td>Good Progressive commits. </td><td></td></tr>
	
  <tr><td colspan='3'><b>Documentation</b></td></tr>
	<tr><td></td><td>Readme :Acceptable. </td><td>-1</td></tr>
	<tr><td></td><td>Model/technical explanation :Good. </td><td></td></tr>
	<tr><td></td><td>Description and Comments :Acceptable. Missing references.</td><td>-1</td></tr>
	<tr><td></td><td>Markdown used and PDF submitted. PDF Checked.</td><td></td></tr>
	
  <tr><td colspan='3'><b>Pull Request</b></td></tr>
	<tr><td></td><td>Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch). </td><td></td></tr>
	<tr><td></td><td>No Feedback required. </td><td></td></tr>
	<tr><td></td><td>Request Description is adequate. </td><td>-1</td></tr>
  <tr><td colspan='2'><b>TOTAL</b></td><td>-6</td></tr>
</tbody></table>

Marked as per the due date and changes after which aren't necessarily allowed to contribute to grade for fairness.
_Subject to approval from Shakes_
